src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -77,26 +76,34 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
 
             try
             {
-                // Get destination type from secret store.
-                DestinationInfo destinationInfo = await GetDestinationInfo(cancellationToken);
+                // Get destination type from secret store and connect to the destination using appropriate client.
+                await GetDestinationInfoAndConnect(cancellationToken);
 
-                // Connect to the destination using appropriate client.
-                _exportDestinationClient = _exportDestinationClientFactory.Create(destinationInfo.DestinationType);
-
-                await _exportDestinationClient.ConnectAsync(destinationInfo.DestinationConnectionString, cancellationToken, _exportJobRecord.Id);
+                // If we are resuming a job, we can detect that by checking the progress info from the job record.
+                // If it is null, then we know we are processing a new job.
+                if (_exportJobRecord.Progress != null)
+                {
+                    // Update our internal mapping with list of files that we have already committed.
+                    foreach (KeyValuePair<string, ExportFileInfo> output in _exportJobRecord.Output)
+                    {
+                        _resourceTypeToFileInfoMapping.Add(output.Key, output.Value);
+                    }
 
-                // TODO: For now, always restart from the beginning. We will support resume in another work item.
-                _exportJobRecord.Progress = new ExportJobProgress(continuationToken: null, page: 0);
+                    await _exportDestinationClient.InitializeForResumingExportAsync(cancellationToken);
+                }
+                else
+                {
+                    _exportJobRecord.Progress = new ExportJobProgress(continuationToken: null, page: 0);
+                }
 
                 ExportJobProgress progress = _exportJobRecord.Progress;
 
-                // Current page will be used to organize a set of search results into a group so that they can be committed together.
+                // Current page will be used to organize a set of search results into a batch so that they can be committed together.
                 uint currentBatchId = progress.Page;
 
-                // The first item is placeholder for continuation token so that it can be updated efficiently later.
                 var queryParameters = new Tuple<string, string>[]
                 {
-                    null,
+                    Tuple.Create(KnownQueryParameterNames.ContinuationToken, progress.ContinuationToken),
```

Comment:
The comment should still be relevant. Maybe just say that the first item will be for continuation token and will be updated as process continues?